### ISSUE TITLE
api->ShowOpenDialog of file_selector freeze UI main thread on windows after start printer

### ISSUE THREAD

**Issue by liqping** [flutter/flutter#123100]
1.file_selector:0.9.2+4 + file_selector_windows-0.9.1+6
2.the ui freezed after start printer;
3.It work fine for run the function in a work thread!

code changed: messages.g.cpp line 180  FileSelectorApi::SetUp
`std::thread([=]{
                  flutter::EncodableMap wrapped;
                  const auto& options_arg = std::any_cast<const SelectionOptions&>(
                      std::get<flutter::CustomEncodableValue>(
                          encodable_options_arg));
                  const auto& encodable_initial_directory_arg = args.at(1);
                  const auto* initial_directory_arg =
                      std::get_if<std::string>(&encodable_initial_directory_arg);
                  const auto& encodable_confirm_button_text_arg = args.at(2);
                  const auto* confirm_button_text_arg =
                      std::get_if<std::string>(&encodable_confirm_button_text_arg);
                  ErrorOr<flutter::EncodableList> output = api->ShowOpenDialog(
                      options_arg, initial_directory_arg, confirm_button_text_arg);
                  if (output.has_error()) {
                      wrapped.emplace(flutter::EncodableValue("error"),
                          WrapError(output.error()));
                  }
                  else {
                      wrapped.emplace(
                          flutter::EncodableValue("result"),
                          flutter::EncodableValue(std::move(output).TakeValue()));
                  }
                  reply(wrapped);
                }).detach();`

**Comment by liqping** [comment]
https://user-images.githubusercontent.com/62126718/226576233-1b6b7b26-5767-4671-a5c7-1613c63a1f28.mp4

**Comment by huycozy** [comment]
Hi @liqping 
For the issue to be workable, it needs to be reproducible with a [completed and minimal reproducible code sample](https://stackoverflow.com/help/minimal-reproducible-example) that doesn’t include 3rd party plugins or complex production code. Please provide such a sample. 

You may also try running [plugin sample code](https://github.com/flutter/packages/tree/main/packages/file_selector/file_selector/example) to check if the issue can reproduce or not. 

Also, please provide the output of `flutter doctor -v` as well. Thanks!

**Comment by liqping** [comment]
flutter doctor -v
`[√] Flutter (Channel stable, 3.3.6, on Microsoft Windows [版本 10.0.19045.2728], locale zh-CN)
    • Flutter version 3.3.6 on channel stable at D:\fluter
    • Upstream repository https://github.com/flutter/flutter.git
    • FLUTTER_GIT_URL = https://github.com/flutter/flutter.git
    • Framework revision 6928314d50 (5 months ago), 2022-10-25 16:34:41 -0400
    • Engine revision 3ad69d7be3
    • Dart version 2.18.2
    • DevTools version 2.15.0
    • Pub download mirror https://pub.flutter-io.cn
    • Flutter download mirror https://storage.flutter-io.cn

[√] Android toolchain - develop for Android devices (Android SDK version 32.0.0)
    • Android SDK at C:\Users\liping\AppData\Local\Android\sdk
    • Platform android-33, build-tools 32.0.0
    • Java binary at: D:\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 11.0.13+0-b1751.21-8125866)
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.13+0-b1751.21-8125866)

[√] Connected device (3 available)
    • Windows (desktop) • windows • windows-x64    • Microsoft Windows [版本 10.0.19045.2728]
    • Chrome (web)      • chrome  • web-javascript • Google Chrome 111.0.5563.65
    • Edge (web)        • edge    • web-javascript • Microsoft Edge 111.0.1661.44

[√] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!
`

**Comment by liqping** [comment]
The code sample: https://github.com/liqping/file_select_test.git

**Comment by liqping** [comment]
The simple_printer_plug is a test printer plug which only pop a windows native dialog to select printer.
PrintDlg see https://learn.microsoft.com/en-us/windows/win32/printdocs/retrieving-a-printer-device-context "Using the PrintDlg Function"

**Comment by liqping** [comment]
https://user-images.githubusercontent.com/62126718/226855974-aeeb262a-b4d1-4da6-8758-0f2539159c96.mp4

**Comment by huycozy** [comment]
Thanks for your response with more information. I check this on my Windows but it works normally. 

https://user-images.githubusercontent.com/104349824/227094581-10c4bf04-074b-4425-83ec-4fab06c392ff.mp4

I'm verifying this on the stable `3.7.7`. Can you check this on the latest stable/master channels?

<details>
<summary>flutter doctor -v (stable & master)</summary>

```bash
[√] Flutter (Channel stable, 3.7.7, on Microsoft Windows [Version 10.0.19045.2604], locale en-US)
    • Flutter version 3.7.7 on channel stable at C:\WIP\flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 2ad6cd72c0 (33 hours ago), 2023-03-08 09:41:59 -0800
    • Engine revision 1837b5be5f
    • Dart version 2.19.4
    • DevTools version 2.20.1

[√] Windows Version (Installed version of Windows is version 10 or higher)

[√] Android toolchain - develop for Android devices (Android SDK version 33.0.0)
    • Android SDK at C:\AndroidSDK
    • Platform android-33, build-tools 33.0.0
    • ANDROID_HOME = C:\AndroidSDK
    • Java binary at: C:\Program Files\Android\Android Studio Electric Eel 2022.1.1\jbr\bin\java
    • Java version OpenJDK Runtime Environment (build 11.0.13+0-b1751.21-8125866)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files (x86)\Google\Chrome\Application\chrome.exe

[√] Visual Studio - develop for Windows (Visual Studio Community 2022 17.4.4)
    • Visual Studio at D:\DOWNLOADWORK\VS2022
    • Visual Studio Community 2022 version 17.4.33213.308
    • Windows 10 SDK version 10.0.19041.0

[√] Android Studio (version 2021.2)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.12+7-b1504.28-7817840)

[√] Android Studio (version 2022.1)
    • Android Studio at C:\Program Files\Android\Android Studio Electric Eel 2022.1.1
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.13+0-b1751.21-8125866)

[√] VS Code (version 1.76.0)
    • VS Code at C:\Users\ADMIN\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.60.0

[√] Connected device (3 available)
    • Windows (desktop) • windows • windows-x64    • Microsoft Windows [Version 10.0.19045.2604]
    • Chrome (web)      • chrome  • web-javascript • Google Chrome 110.0.5481.180
    • Edge (web)        • edge    • web-javascript • Microsoft Edge 110.0.1587.41

[√] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!
```

```bash
[!] Flutter (Channel master, 3.9.0-7.0.pre.33, on Microsoft Windows [Version 10.0.19045.2604], locale en-US)
    • Flutter version 3.9.0-7.0.pre.33 on channel master at C:\WIP\flutter_master
    ! Warning: `flutter` on your path resolves to C:\WIP\flutter\bin\flutter, which is not inside your current
      Flutter SDK checkout at C:\WIP\flutter_master. Consider adding C:\WIP\flutter_master\bin to the front of your
      path.
    ! Warning: `dart` on your path resolves to C:\WIP\flutter\bin\dart, which is not inside your current Flutter SDK
      checkout at C:\WIP\flutter_master. Consider adding C:\WIP\flutter_master\bin to the front of your path.
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision fb7e82826d (2 hours ago), 2023-03-15 04:34:21 -0400
    • Engine revision b7297f0259
    • Dart version 3.0.0 (build 3.0.0-322.0.dev)
    • DevTools version 2.22.2
    • If those were intentional, you can disregard the above warnings; however it is recommended to use "git"
      directly to perform update checks and upgrades.

[√] Windows Version (Installed version of Windows is version 10 or higher)

[√] Android toolchain - develop for Android devices (Android SDK version 33.0.0)
    • Android SDK at C:\AndroidSDK
    • Platform android-33, build-tools 33.0.0
    • ANDROID_HOME = C:\AndroidSDK
    • Java binary at: C:\Program Files\Android\Android Studio Electric Eel 2022.1.1\jbr\bin\java
    • Java version OpenJDK Runtime Environment (build 11.0.13+0-b1751.21-8125866)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files (x86)\Google\Chrome\Application\chrome.exe

[√] Visual Studio - develop for Windows (Visual Studio Community 2022 17.4.4)
    • Visual Studio at D:\DOWNLOADWORK\VS2022
    • Visual Studio Community 2022 version 17.4.33213.308
    • Windows 10 SDK version 10.0.19041.0

[√] Android Studio (version 2021.2)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.12+7-b1504.28-7817840)

[√] Android Studio (version 2022.1)
    • Android Studio at C:\Program Files\Android\Android Studio Electric Eel 2022.1.1
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.13+0-b1751.21-8125866)

[√] VS Code (version 1.76.1)
    • VS Code at C:\Users\ADMIN\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.60.0

[√] Connected device (3 available)
    • Windows (desktop) • windows • windows-x64    • Microsoft Windows [Version 10.0.19045.2604]
    • Chrome (web)      • chrome  • web-javascript • Google Chrome 111.0.5563.65
    • Edge (web)        • edge    • web-javascript • Microsoft Edge 110.0.1587.41

[√] Network resources
    • All expected network resources are available.

! Doctor found issues in 1 category.
```
</details>
